### DISCUSSION TITLE
Allow arbitrary user names for RLS?

### DISCUSSION THREAD

**Discussion by CRSnyder** [DaxStudio/DaxStudio#656] [Q&A]
We're writing app-owned embedded reports and using row-level security to restrict what a user sees.   With app-owned embedded reports, the user name used by RLS can be anything, not just an email address.  We are using user ids as the user name and I'm hoping to use Dax Studio to assist in figuring out how to apply the correct RLS rules.  However, when I used an id as Effective User Name, I received the error, "The name provided is not a properly formed account name."  I assume this is because the user name isn't an email address or AD account.  Using an arbitrary name is possible in PowerBI Desktop (though not on powerbi.com). 

![image](https://user-images.githubusercontent.com/2047777/123168202-8205cd80-d42c-11eb-9a8b-3d24477f0493.png)

 Is doing something similar possible in Dax Studio, or would it be feasible to implement? My data source is a local pbix file.  Having the ability to have user names of any format would be really handy!

Thanks.

**Comment by dgosbell** [comment]
I've had a bit of a look and Power BI is using some sort of alternative authentication scheme that I have not seen documented anywhere to achieve this. I'll need to do some research to find out if there is any way for third party tools to leverage this same mechanism.

**Reply by CRSnyder** [reply]
Thank you so much for looking into this!
